src/main/java/com/kodilla/royal/game/of/ur/Game.java

```diff
@@ -0,0 +1,273 @@
+
+    private final static int ROW_COUNT = 3;
+    private final static int COLUMN_COUNT = 8;
+
+    private int diceRollResult;
+    private Player currentPlayer;
+    private final Dice dice;
+
+    private final HumanPlayer humanPlayer;
+    private boolean canHumanPickPiece;
+    private final ComputerPlayer computerPlayer;
+    private final BoardController boardController;
+
+    public Game() throws Exception {
+        Field[][] fields = new Field[ROW_COUNT][COLUMN_COUNT];
+        for (int row = 0; row < ROW_COUNT; row++) {
+            for (int col = 0; col < COLUMN_COUNT; col++) {
+                if (row % 2 == 1 || col < 2 || col > 3) {
+                    fields[row][col] = new Field(row, col, !(row % 2 == 0 || col == 4));
+                }
+            }
+        }
+
+        boardController = new BoardController(this);
+
+        currentPlayer = humanPlayer = new HumanPlayer(this, Route.SIMPLE_ROUTE, fields);
+        canHumanPickPiece = false;
+        computerPlayer = new ComputerPlayer(this, Route.SIMPLE_ROUTE, fields);
+
+        dice = new Dice();
+//        dice = new Dice(4);
+
+        boardController.setNewGameButtonAction(event -> {
+            String message = "Do you really want to start a new game?\nYou will loose your current game progress.";
+            if (boardController.showConfirmationDialog(message)) {
+                resetGame();
+            }
+        });
+        boardController.setInstructionsButtonAction(event -> {
+            List<String> rawInstructions = retrieveGameInstructions();
+            List<List<String>> instructions = prepareGameInstructions(rawInstructions);
+            showInstructions(instructions);
+        });
+
+        boardController.setHumanPiecesMouseClickAction(this::handlePieceMouseClicked);
+        boardController.setHumanPiecesMouseEnterAction(this::handlePieceMouseEntered);
+        boardController.setHumanPiecesMouseExitAction(this::handlePieceMouseExited);
+        boardController.setRollDiceButtonAction(event -> handleRollDiceButtonClick());
+
+        boardController.setTurnIndicator(humanPlayer, GREEN_LIGHT);
+        boardController.setTurnIndicator(computerPlayer, RED_LIGHT);
+
+//        capture test
+//        humanPlayer.setPieceAtField(0, 4);
+//        computerPlayer.setPieceAtField(0, 7);
+
+//        finish game test
+//        humanPlayer.setPieceAtField(0, 12);
+//        computerPlayer.setPieceAtField(0, 12);
+
+//        cant move test
+//        humanPlayer.setPieceAtField(0, 5);
+//        humanPlayer.setPieceAtField(1, 8);
+
+//        no moves available
+//        humanPlayer.setPieceAtField(0, 5);
+//        computerPlayer.setPieceAtField(0, 8);
+
+//        Field field = fields[0][5];
+//        boardController.highlightField(field, RED_HIGHLIGHT);
+    }
+
+    public BoardController getBoardController() {
+        return boardController;
+    }
+
+    private void handlePieceMouseClicked(MouseEvent event) {
```

Comment:
wyciagnalbym to do innej klasy

Reply:
Mógłbym utworzyć nową klasę, np. `EventHandlers` i przenieść do niej wszystkie metody odpowiedzialne za obsługę zdarzeń. Problem mam z tym, że metody te używają szeregu pól klasy `Game`. O ile z obiektami nie byłoby większego problemu, bo mógłbym je wstrzyknąć do nowej klasy za pośrednictwem konstruktora, o tyle pola `canHumanPickPiece` i `diceRollResult` są typów prymitywnych i wstrzyknąć się ich nie da. Mógłbym pewnie przenieść je również do nowej klasy i operaować na nich za pomocą getterów i setterów lub użyć typów opakowujących `Boolean` i `Integer`. A może jest jeszcze inne rozwiązanie?